src/ui/CategoryFacet/CategoryFacetSearch.ts

```diff
@@ -38,17 +38,18 @@ export class CategoryFacetSearch implements IFacetSearch {
   public build() {
     this.container = $$('div', {
       className: 'coveo-category-facet-search-container',
-      role: 'heading',
-      'aria-level': 3
+      role: 'button'
     });
 
+    const title = this.categoryFacet.options.title;
+
     new AccessibleButton()
       .withElement(this.container)
       .withSelectAction(() => {
         $$(this.categoryFacet.element).addClass('coveo-category-facet-searching');
         this.focus();
       })
-      .withLabel(l('Search'))
+      .withLabel(title ? l('SearchFacetResults', this.categoryFacet.options.title) : l('Search'))
```

Comment:
When there is no title, why not use the `SearchFacetResults` message? Wouldn't the `Search` message be a little abstract?

Reply:
Good point, I'll put a less abstract message.